include: zephyr: drivers: espi: Adjust terms to match eSPI specification 1.5

albertofloyd opened the pull request:
Replace master/slave by new controller/target terminology in eSPI spec 1.5.

Propagate eSPI VW change and macro name changes across eSPI drivers
Propagate eSPI VW changes to eSPI driver sample and eSPI test driver code
Add support for MEC172x modular card for samples/drivers/espi to sanity check changes

Changed scope of PR to ease review, the eSPI API clarifications moved to https://github.com/zephyrproject-rtos/zephyr/pull/69358

albertofloyd committed:
doc: eSPI: Update link to eSPI spec

Add link to eSPI spec 1.5
Update coding guidelines for eSPI bus

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

keith-zephyr commented on drivers/espi/espi_mchp_xec.c:
```diff
@@ -113,107 +113,107 @@ enum mchp_smvw_regs {
 	/* MSVW00 */
 	[ESPI_VWIRE_SIGNAL_SLP_S3]        = {MCHP_MSVW00, ESPI_VWIRE_SRC_ID0,
-					     ESPI_MASTER_TO_SLAVE},
+					     ESPI_CONTROLLER_TO_TARGET},
```
This commit should be squashed with the change the header file.  Otherwise this will break bi-sectability.

albertofloyd replied:
Commits squashed

keith-zephyr reviewed (commented):
Happy to see this change.  Can you also update https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html?

albertofloyd commented:
> Happy to see this change. Can you also update https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html?

Updated the coding guidelines too

keith-zephyr commented on samples/drivers/espi/boards/mec172xmodular_assy6930.conf:
```diff
@@ -0,0 +1,10 @@
+# eSPI + mec172xevb_assy6906
```
This new addition to the eSPI samples should be split into a separate commit.

albertofloyd replied:
Moved the config/overlay files into separate commit

albertofloyd committed:
treewide: drivers: espi: Adjust terms per eSPI specification 1.5

1) Replace master/slave in API for new terms in eSPI spec 1.5
2) Reflect eSPI VW change and macro changes across eSPI drivers
3) Update terms in eSPI driver sample and eSPI test driver

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

albertofloyd committed:
samples: drivers: espi: Add support for mec172x board

Add overlay file for mec172x modular card

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

aescolar commented:
@albertofloyd is this an api change which requires discussion in the arch wg? (CC @carlescufi )

dleach02 reviewed (changes requested):
Thank you for doing this. Outside of the question about API change from @aescolar, if this is okay then this PR needs to include the change information in the migration guide (and release notes updated)

ChiHuaL commented:
Sorry for the late comment. 
You missed to update the inclusive terms for the file include/zephyr/drivers/espi_saf.h
Also, do you plan to update term `saf` to `taf` and `maf` to `caf`?

albertofloyd commented:
> Sorry for the late comment. You missed to update the inclusive terms for the file include/zephyr/drivers/espi_saf.h Also, do you plan to update term `saf` to `taf` and `maf` to `caf`?

Yes, plan is to update also eSPI SAF drivers API to move to CAF in separate PR

albertofloyd commented:
> Thank you for doing this. Outside of the question about API change from @aescolar, if this is okay then this PR needs to include the change information in the migration guide (and release notes updated)

Ack'd

albertofloyd commented:
> o update also eSPI SAF drivers API to move to CAF in separate PR

> @albertofloyd is this an api change which requires discussion in the arch wg? (CC @carlescufi )

@aescolar @carlescufi  I don't think so, mostly delta between 1.0 and 1.5 are ECN not applicable to Zephyr eSPI driver except the inclusive language.  
 • Merged eSPI Server Addendum Rev0.7 into the eSPI Base specification. Target Attached Flash sharing is applicable for client and server platforms.
• Included ECN- Target Attached Flash RPMC.
• Included ECN- Clarify Flash Erase Length (May 2020)
• Included ECN- BMC integrated RTC (7-11-2020).
• Updated the terms “master/slave” to “controller/target” throughout the specification to align with the use of inclusive language.

albertofloyd committed:
doc: Update eSPI sections in migration and release notes

Indicate macro and enum changes with respect to new inclusive
term in eSPI specification 1.5.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

kartben commented on doc/releases/migration-guide-3.7.rst:
```diff
@@ -120,6 +120,14 @@ Controller Area Network (CAN)
 Display
 =======
 
+Enhanced Seria Peripheral Interface (eSPI)
+==========================================
```
```suggestion
Enhanced Serial Peripheral Interface (eSPI)
===========================================
```

albertofloyd replied:
Fixed

kartben commented on doc/releases/migration-guide-3.7.rst:
```diff
@@ -120,6 +120,14 @@ Controller Area Network (CAN)
 
+Enhanced Seria Peripheral Interface (eSPI)
+==========================================
+  * The macros ESPI_SLAVE_TO_MASTER and ESPI_MASTER_TO_SLAVE were renamed to ESPI_TARGET_TO_CONTROLLER
+    and ESPI_CONTROLLER_TO_TARGET respectively to reflect new terminology in eSPI 1.5 specification.
+  * The enum values ESPI_VWIRE_SIGNAL_SLV_BOOT_STS, ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE and ESPI_VWIRE_SIGNAL_SLV_GPIO_XX
+    were renamed to ESPI_VWIRE_SIGNAL_TARGET_BOOT_STS, ESPI_VWIRE_SIGNAL_TARGET_BOOT_DONE and
+    ESPI_VWIRE_SIGNAL_TARGET_GPIO_XX to reflect new terminology in eSPI 1.5 specification.
```
consider enclosing code litterals between double backquotes.

albertofloyd replied:
Done

kartben reviewed (changes requested):
btw, are there plans to graduate eSPI to a Stable API?

kartben commented on doc/releases/release-notes-3.7.rst:
```diff
@@ -120,6 +120,17 @@ Drivers and Sensors
 
+* eSPI
+
+  * The macros ``ESPI_SLAVE_TO_MASTER`` and ``ESPI_MASTER_TO_SLAVE`` were renamed to
+    ``ESPI_TARGET_TO_CONTROLLER`` and ``ESPI_CONTROLLER_TO_TARGET`` respectively to reflect
+    new terminology in eSPI 1.5 specification.
+  * The following virtual enum values ``ESPI_VWIRE_SIGNAL_SLV_BOOT_STS``,
+    ``ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE`` and ``ESPI_VWIRE_SIGNAL_SLV_GPIO_<NUMBER>`` were renamed to
+    ``ESPI_VWIRE_SIGNAL_TARGET_BOOT_STS``, ``ESPI_VWIRE_SIGNAL_TARGET_BOOT_DONE`` and
+    ``ESPI_VWIRE_SIGNAL_TARGET_GPIO_<NUMBER>`` respectively to reflect new terminology
+    in eSPI 1.5 specification.
```
this shouldn't be paraphrasing the migration guide entry. Either drop altogether or, probably better, just make it a simpler bullet point "Renamed several eSPI APIs to adopt terminology from version 1.5 of the specification" or something like this?

albertofloyd replied:
Summarized

kartben reviewed (changes requested):
this is really great, thanks for doing this. I have a minor request for change re: release notes entry. Thanks!

albertofloyd commented:
@kartben updated release notes

scottwcpg commented:
@dleach02 @albertofloyd Hi, can we please get this merged? I have a new SoC eSPI driver and would like to have this done before submitting the new driver. Please, please, please...

fabiobaltieri commented:
@dleach02 ping, you have a blocker on this

albertofloyd commented:
ping @dleach02

kartben commented:
@albertofloyd what's the plan with Kconfig `CONFIG_ESPI_SLAVE`, shouldn't it be renamed to `CONFIG_ESPI_TARGET`?
Thanks again for this work!

albertofloyd commented:
> @albertofloyd what's the plan with Kconfig `CONFIG_ESPI_SLAVE`, shouldn't it be renamed to `CONFIG_ESPI_TARGET`? Thanks again for this work!

Working in this PR https://github.com/zephyrproject-rtos/zephyr/pull/73887
